src/web-components/vf-score.js

```diff
@@ -0,0 +1,137 @@
+// ## Description
+// 
+// This file implements `vf-score`, the web component that acts as the 
+// container for the entire component. `vf-score`'s shadow root holds
+// the HTML element that the renderer renders too.
+// 
+// All actual drawing is called from `vf-score`. 
+
+import Vex from '../index';
+
+export class VFScore extends HTMLElement {
+  constructor() {
+    super();
+
+    // Defaults
+    this.x = 10; // The starting x position of a system within the score.
+    this.y = 0; // The starting y position of system within the score. 
+    this.width = 500; // The entire width of the element holding the music notation. 
+    this.height = 150; // The entire height of the element holding the music notation.
+    this.stavesPerLine = 1; 
+
+    this.systemsAdded = 0; // Counter that keeps track of how many systems have dispatched 
+                           // events signalling that they are ready to be drawn. When the
+                           // number of children matches this counter, the entire score
+                           // is ready to be drawn.
+                           // FOR THIS PR: a vf-score can only have one vf-system. 
+
+    this.attachShadow({ mode:'open' });
+
+    // The 'systemCreated' event is dispatched by a vf-systen when it has finished creating 
+    // and adding its staves. vf-score listens to this event so that it can add that it can 
+    // detect when the vf-system is ready to be drawn.
+    this.addEventListener('systemCreated', this.systemCreated);
+
+    // These events are dispatched by the corresponding elements (ex: vfVoiceReady is dispatched
+    // by a vf-voice) when they are added to the DOM. vf-score listens to these events so that it 
+    // can set the elements' Factory and/or Registry instances, since these are shared by a 
+    // vf-score and all its children to maintain the same render queue. 
+    this.addEventListener('vfVoiceReady', this.setFactory);
+    this.addEventListener('vfStaveReady', this.setFactory);
+    this.addEventListener('vfStaveReady', this.setRegistry);
+    this.addEventListener('vfSystemReady', this.setFactory);
+  }
+
+  connectedCallback() {
+    this.x = parseInt(this.getAttribute('x')) || this.x;
+    this.y = parseInt(this.getAttribute('y')) || this.y;
+    this.width = parseInt(this.getAttribute('width')) || this.width;
+    this.height = parseInt(this.getAttribute('height')) || this.height;
+
+    this.rendererType = this.getAttribute('renderer'); 
+
+    this.setupVexflow();
+    this.setupFactory();
+
+    // vf-score listens to the slotchange event so that it can detect its system and set it up accordingly
+    this.shadowRoot.querySelector('slot').addEventListener('slotchange', this.registerSystem);
+  }
+
+  disconnectedCallback() {
+    this.shadowRoot.querySelector('slot').removeEventListener('slotchange', this.registerSystem);
```

Comment:
If you do setup work in `connectedCallback` you might need to cleanup in `disconnectedCallback`, ie release VexFlow and Factory

Reply:
I ended up moving the setup functions to the constructor, but what would "releasing" VexFlow and Factory look like?

Reply:
It depends on if the VexFlow object held onto any resources that need to be released. It may not, but it could have event handlers on `window` that would cause the VexFlow object from being GC'ed. Usually if an object needs to be cleaned up it'll have something like a `destroy()` or `cleanup()` method.